lib/active_record_shards/deprecation.rb

```diff
@@ -6,6 +6,7 @@ class Deprecation < ActiveSupport::Deprecation
     # For example, this allows us to silence our own deprecation warnings in test while still being
     # able to fail tests for upstream deprecation warnings.
     def behavior
+      ActiveSupport::Deprecation.behavior = :log
```

Comment:
@craig-day without setting this log behaviour, we found that deprecated methods were raising errors in test and development. happy to keep this for this gem?

ref:
> Deprecation warnings raised by gems are not affected by this setting because they happen before Rails boots up.
https://edgeapi.rubyonrails.org/classes/ActiveSupport/Deprecation/Behavior.html#method-i-behavior-3D

Reply:
Will it still be possible to configure it to use e.g. `:raise`?

Reply:
removing this, I was incorrect, in a Rails app `config.active_support.deprecation = :log` works as expected.